packages/functions/src/dataTriggers.ts

```diff
@@ -269,7 +269,82 @@ export const createAttendanceForBooking = functions
 
     await attendanceRef.set(updatedEntry, { merge: true });
   });
+// SlotAttendnace = {
+//   /**
+//    * ISO date of the slot for easier querying
+//    */
+//   date: string;
+//   /**
+//    * Attendances for slot keyed by customer
+//    */
+//   attendances: {
+//     [customerId: string]: CustomerAttendance;
+//   };
+// };
+
+// CustomerAttendance {
+//   bookedInterval: string | null;
+//   attendedInterval: string | null;
+// }
+
+// entry
+// [slotId: string]: CustomerBookingEntry;
+// {
+//   /**
+//    * ISO date of the booked slot (used for easier querying from store)
+//    */
+//   date: string;
+//   /**
+//    * Booked interval for particular slot
+//    */
+//   interval: string;
+
+// }
+export const createAttendedSlotOnAttendance = functions
+  .region(__functionsZone__)
+  .firestore.document(
+    `${Collection.Organizations}/{organization}/${OrgSubCollection.Attendance}/{bookingId}/`
+  )
+  .onWrite(async (change, context) => {
+    const { organization, bookingId } = context.params as Record<
+      string,
+      string
+    >;
+
+    const db = admin.firestore();
+
+    const isCreate = !change.before.exists;
+    // const isDelete = !change.after.exists;
```

Comment:
Ok...so here, it would be useful to get `isCreate` and `isUpdate`. If it's create, you would update each customer's attended slots, and on update, only the updated (that can be achieved by comparing `before` and `after` data)